src/core/Xtarget.ml

```diff
@@ -1,16 +1,56 @@
-(* eXtended target.
+(* Yoann Padioleau, Cooper Pierce
  *
- * This type is mostly used in the engine, to pass around extra information
- * associated to each target.
+ * Copyright (c) Semgrep Inc.
  *
- * related: Input_to_core.target, which is what is passed
- * to semgrep-core via -target.
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * version 2.1 as published by the Free Software Foundation, with the
+ * special exception on linking described in file LICENSE.
+ *
+ * This library is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the file
+ * LICENSE for more details.
+ *)
+
+(** eXtended target.
+
+   This type is mostly used in the engine to pass around extra information
+   (e.g., contents, the AST) associated with each {{!Target.code}target}.
+
+   See also {!Input_to_core_t.target}, which is what is passed to
+   [semgrep-core] via [-target].
  *)
 
 type t = {
-  file : Fpath.t;
-  xlang : Xlang.t;
+  path : Target.path;
```

Comment:
at this point maybe we could wrap completely Target.t instead of Target.path,
since Target.t contains also xlang that we need for Xtarget.

Reply:
The unfortunate issue there is that you then must carry around the `Target.lockfile`, if one is associated. Ideally we would carry the `Lockfile_xtarget.t` around instead (but in any case, we do not currently need to access it from the `Xtarget` and so such a field does not exist here).

I can refactor it to do that, but I anticipate it would simply add to memory consumption for rather marginal benefit.